src/base/web3-connection.ts

```diff
@@ -10,7 +10,12 @@ export class Web3Connection {
   protected web3!: Web3;
   protected account!: Account;
 
-  constructor(readonly options: Web3ConnectionOptions) {}
+  constructor(readonly options: Web3ConnectionOptions) {
+    // If a provider is supplied, we assume all connection logic is on its side.
+    if (options?.provider) {
```

Comment:
We can't assume the user wants to auto-start when creating a new Web3Connection as they might want to provide the returned value as a singleton to the whole application - starting it _ourselves_ would probably go against the users flow.

Reply:
removed as agreed on discord